integrationTests/testHeartbeatNode.go

```diff
@@ -495,6 +495,7 @@ func (thn *TestHeartbeatNode) initInterceptors() {
 		CoreComponents: &processMock.CoreComponentsMock{
 			IntMarsh: TestMarshaller,
 		},
+		ShardCoordinator:             thn.ShardCoordinator,
```

Comment:
instead of using comments on L506, L515, L526, you could have extracted the code in simple functions. 
Bonus if you apply this: all defined arguments can be called `args` as they "live" only inside those functions.